gateway/ln-gateway/src/lib.rs

```diff
@@ -236,6 +229,10 @@ pub struct Gateway {
 
     /// The task group for all tasks related to the gateway.
     task_group: TaskGroup,
+
+    /// The bcrypt password hash used to authenticate the gateway.
+    /// This is an `Arc` because `bcrypt::HashParts` does not implement `Clone`.
+    bcrypt_password_hash: Arc<bcrypt::HashParts>,
```

Comment:
Is this necessary here? Can you just pass `bcrypt_password_hash` as a parameter to the rpc server?

Reply:
This would require moving where we parse the gateway options:

https://github.com/fedimint/fedimint/blob/7b7bf0f45574911e8348c68f2abc5c61b36b117c/gateway/ln-gateway/src/lib.rs#L296-L297

We'd need to move it up a level:

https://github.com/fedimint/fedimint/blob/7b7bf0f45574911e8348c68f2abc5c61b36b117c/gateway/ln-gateway/src/bin/gatewayd.rs#L26-L38

And pass it into `.run()`

I can do this as a follow-up, but this seems outside the scope of this PR.